app/core/pokemon-entity.ts

```diff
@@ -1251,6 +1251,29 @@ export class PokemonEntity extends Schema implements IPokemonEntity {
       this.addDefense(5, this, 0, false)
       this.addSpecialDefense(5, this, 0, false)
     }
+
+    if ((this.passive === Passive.ZYGARDE10 || this.passive === Passive.ZYGARDE50) && this.life < 0.5 * this.hp) {
+
+      if (this.passive === Passive.ZYGARDE10){
+        this.addDefense(1, this, 0, false)
+        this.addSpecialDefense(1, this, 0, false)
+        this.addPP(20, this, 0, false)
+        this.addMaxHP(50, this, 0, false)
+      } else {
+        this.addAttack(5, this, 0, false)
+        this.addDefense(2, this, 0, false)
+        this.addSpecialDefense(2, this, 0, false)
+        this.addMaxHP(80, this, 0, false)
+        this.range = 1
```

Comment:
need to take into account wide lens. Use range = min(1)(range - difference) instead